src/c/core/session/session.c

```diff
@@ -331,10 +367,14 @@ bool wait_session_status(uxrSession* session, uint8_t* buffer, size_t length, si
     return session->info.last_requested_status != UXR_STATUS_NONE;
 }
 
-inline void send_message(const uxrSession* session, uint8_t* buffer, size_t length)
+inline bool send_message(const uxrSession* session, uint8_t* buffer, size_t length)
 {
-    (void) session->comm->send_msg(session->comm->instance, buffer, length);
-    UXR_DEBUG_PRINT_MESSAGE(UXR_SEND, buffer, length, session->info.key);
+    bool sent = session->comm->send_msg(session->comm->instance, buffer, length);
+    if (sent)
+    {
+        UXR_DEBUG_PRINT_MESSAGE(UXR_SEND, buffer, length, session->info.key);
```

Comment:
Maybe it could be better to print an error trace instead of the message?

Reply:
I think that the message log will be always printed. But could be cool to put a message error when the transport returns a known error. Even, if the type of this error can be showed to the user.

The main reason because I always let the log to print the message is because the error is highly dependent on the transport type and the platform implementation type. Could have messages that, at the same error, be printed or not depends on the chosen transport/implementation/platform. The user can be confused if the message is not arriving but sometimes the message is printed and sometimes not.

Reply:
Ok, I added two new debug traces for messages which saw the messages in case of error at writing and reading.